crates/turbopack-core/src/chunk/mod.rs

```diff
@@ -124,26 +124,41 @@ impl ChunkGroupVc {
     /// All chunks should be loaded in parallel.
     #[turbo_tasks::function]
     pub async fn chunks(self) -> Result<ChunksVc> {
-        let mut chunks = IndexSet::new();
-
-        let mut queue = vec![self.await?.entry];
-        while let Some(chunk) = queue.pop() {
-            let chunk = chunk.resolve().await?;
-            if chunks.insert(chunk) {
-                for r in chunk.references().await?.iter() {
-                    if let Some(pc) = ParallelChunkReferenceVc::resolve_from(r).await? {
-                        if *pc.is_loaded_in_parallel().await? {
-                            let result = r.resolve_reference();
-                            for a in result.primary_assets().await?.iter() {
-                                if let Some(chunk) = ChunkVc::resolve_from(a).await? {
-                                    queue.push(chunk);
-                                }
+        let chunks = [self.await?.entry]
+            .into_iter()
+            .try_flat_map_recursive_join(|chunk| async move {
+                Ok(chunk
```

Comment:
Split up these huge method chains into intermediate items for readability purposes imo.

```rust
                let references = chunk
                    .references()
                    .await?;
                let parallel_chunks = references.iter().map(|r| ...);
```
etc.

Same comment for https://github.com/vercel/turbo/pull/3619/files#diff-920fc3f07551ceb49ecd59a0730dae802cb4f7b906b576defad3149d50933deeR27 and https://github.com/vercel/turbo/pull/3619/files#diff-b6de4264e8fc5bf158812dc1ebd1697428cb28e16adad1eda0cdd339e96b9151R138

While rust-analyzer might show the intermediate types, it doesn't show on GH and the flow of the transformation is unclear.